pyPhin.py

```diff
@@ -165,37 +229,167 @@ def getWaterData(self, authToken, deviceUUID, vesselUrl):
 		self.checkRequest(req)
 		reqJson = json.loads(req.text)
 
-		data = {}
+		data = {"waterData":{},"pool":{}}
 
 		for dataType in ["TA","CYA","TH"]:
 			try:
-				data[dataType] = reqJson["vessels"][0]["waterReport"][dataType]["value"]
+				data["waterData"][dataType.lower()] = reqJson["vessels"][0]["waterReport"][dataType]["value"]
 			except:
 				self.logger.error("Not able to access %s with %s",dataType,req.text)
 
 		try:
-			data["temperature"] = reqJson["vessels"][0]["disc"]["temperatureF"]
+			data["waterData"]["temperature"] = reqJson["vessels"][0]["disc"]["temperatureF"]
 		except:
 			self.logger.error("Not able to access temperature with %s",req.text)
 		try:
-			data["status"] = reqJson["vessels"][0]["disc"]["title"]
+			data["pool"]["status_title"] = reqJson["vessels"][0]["disc"]["name"]
 		except:
 			self.logger.error("Not able to access temperature with %s",req.text)
 		try:
-			data["status_id"] = reqJson["vessels"][0]["disc"]["waterStatus"]["value"]
+			data["pool"]["status_id"] = reqJson["vessels"][0]["disc"]["waterStatus"]["value"]
 		except:
-			self.logger.error("Not able to access status ID with %s",req.text)
+			self.logger.error("Not able to access status id with %s",req.text)
+
+		chartData = self.getChartData(
+			authToken,
+			deviceUUID,
+			reqJson["vessels"][0]["widgets"][0]["resources"]["appChartsWeek"]["route"])
+
+		returnData = [data,chartData]
+
 
 		'''Sample Return data
+		[
+			{
+				"waterData":{
+					"ta":100,
+					"cya": 40,
+					"th": 170,
+					"temperature":70.9
+				}
+				"pool":{
+					"status_title": "balanced",
+					"status_id": 1
+					}
+			},
+			{
+				"waterData":{
+					"ph":{"value":8.2, "status":1},
+					"orp":{"value":800,"status":2}
+				}
+				"vesselData":{
+					"battery":{"value":2000, "percentage":0.80},
+					"rssi":{"value":-91,"status":3}
+				}
+			}
+		]
+		'''
+		return returnData
+
+	def getChartData(self, authToken, deviceUUID, chartUrl):
+		req = requests.get(
+			self.baseUrl + chartUrl,
+			headers=self.createHeader(deviceUUID, authToken, "1.0.0")
+			)
+		self.checkRequest(req)
+		reqJson = json.loads(req.text)
+
+		chartData = {"waterData":{},"vesselData":{}}
+
+		'''Status Codes
+		1 - Needs Immediate Attention Low
+		2 - Needs Attention Low
+		3 - Ok
+		4 - Needs Attention High
+		5 - Needs Immediate Attention High
+		'''
+
+		#PH
+		try:
+			phAvg = sum(reqJson["ph"][-self.phDataPointAvgLen:])/self.phDataPointAvgLen
+
+			phStatus = -1
+			if phAvg < 6.8:
+				phStatus = 1
+			elif phAvg < 7:
+				phStatus = 2
+			elif phAvg >= 7 and phAvg <= 7.5:
+				phStatus = 3
+			elif phAvg > 7.5:
+				phStatus = 4
+			elif phAvg > 7.8:
```

Comment:
it can never hit status == 5 as phAve > 7.5 will set it to 4... work backwards, eg start with phAvg > 7.8